### PR TITLE
Fix a crash when trying to build very large Reed-Muller codes

### PR DESCRIPTION
**Pull request by jlavauzelle**
This PR aims at fixing #33229. The problem was that, when trying to build a very large Reed-Muller code, the constructor called from the `AbstractLinearCodeNoMetric` parent class was building an element of the code. In the given crashing example, the size of this element (4294967296) probably exceeds the maximal size for memory allocation, leading to segfault.

The problem is simply fixed by **not building** this element, since in the constructor, we actually only need the type of the element.

I added an example in the `BinaryReedMuller` class to emphasize this new working behaviour.

### :memo: Checklist

- [x] The title is concise and informative.
- [x] The description explains in detail what this PR is about.
- [x] I have linked a relevant issue or discussion.
- [x] I have created tests covering the changes.
- [x] I have updated the documentation and checked the documentation preview.

### :hourglass: Dependencies

#33229: where the issue was raised.

### CODE REVIEW

File: src/sage/coding/reed_muller_code.py [hunk 1]
```diff
@@ -401,6 +401,14 @@ class directly, as :meth:`ReedMullerCode` creates either a binary or a
         sage: C = codes.BinaryReedMullerCode(2, 4)
         sage: C
         Binary Reed-Muller Code of order 2 and number of variables 4
+
+    Very large Reed-Muller codes can be constructed without building the generator matrix or elements of the code.
```
Comment (dcoudert) [inline src/sage/coding/reed_muller_code.py]:
```diff
-    Very large Reed-Muller codes can be constructed without building the generator matrix or elements of the code.
+    Very large Reed-Muller codes can be constructed without building
+    the generator matrix or elements of the code::
```

**Review by dcoudert — COMMENTED** [review]
otherwise LGTM.

File: src/sage/coding/reed_muller_code.py (hunk 1 above)
Comment (grhkm21) [reply src/sage/coding/reed_muller_code.py]: Also refer to the issue number ``:issue:`39110` `` (without the space).

Comment (jlavauzelle) [reply src/sage/coding/reed_muller_code.py]: Thanks for your reviews. I fixed the doc.

**Review by dcoudert — APPROVED** [review]
LGTM.

**Pull request merged**